cmd/podman/main_local.go

```diff
@@ -0,0 +1,154 @@
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.LogLevel, "log-level", "error", "Log messages above specified level: debug, info, warn, error, fatal or panic")
+	rootCmd.PersistentFlags().IntVar(&MainGlobalOpts.MaxWorks, "max-workers", 0, "The maximum number of workers for parallel operations")
+	rootCmd.PersistentFlags().MarkHidden("max-workers")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Namespace, "namespace", "", "Set the libpod namespace, used to create separate views of the containers and pods on the system")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Root, "root", "", "Path to the root directory in which data, including images, is stored")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Runroot, "runroot", "", "Path to the 'run directory' where all state information is stored")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Runtime, "runtime", "", "Path to the OCI-compatible binary used to run containers, default is /usr/bin/runc")
+	// -s is depracated due to conflict with -s on subcommands
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.StorageDriver, "storage-driver", "", "Select which storage driver is used to manage storage of images and containers (default is overlay)")
+	rootCmd.PersistentFlags().StringSliceVar(&MainGlobalOpts.StorageOpts, "storage-opt", []string{}, "Used to pass an option to the storage driver")
+	rootCmd.PersistentFlags().BoolVar(&MainGlobalOpts.Syslog, "syslog", false, "Output logging information to syslog as well as the console")
+
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.TmpDir, "tmpdir", "", "Path to the tmp directory")
+	rootCmd.PersistentFlags().BoolVar(&MainGlobalOpts.Trace, "trace", false, "Enable opentracing output")
+}
+
+func setSyslog() error {
+	if MainGlobalOpts.Syslog {
+		hook, err := lsyslog.NewSyslogHook("", "", syslog.LOG_INFO, "")
+		if err == nil {
+			logrus.AddHook(hook)
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+func profileOn(cmd *cobra.Command) error {
+	if cmd.Flag("cpu-profile").Changed {
+		f, err := os.Create(MainGlobalOpts.CpuProfile)
+		if err != nil {
+			return errors.Wrapf(err, "unable to create cpu profiling file %s",
+				MainGlobalOpts.CpuProfile)
+		}
+		if err := pprof.StartCPUProfile(f); err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flag("trace").Changed {
+		var tracer opentracing.Tracer
+		tracer, closer = tracing.Init("podman")
+		opentracing.SetGlobalTracer(tracer)
+
+		span = tracer.StartSpan("before-context")
+
+		Ctx = opentracing.ContextWithSpan(context.Background(), span)
+	}
+	return nil
+}
+
+func profileOff(cmd *cobra.Command) error {
+	if cmd.Flag("cpu-profile").Changed {
+		pprof.StopCPUProfile()
+	}
+	if cmd.Flag("trace").Changed {
+		span.Finish()
+		closer.Close()
+	}
+	return nil
+}
+
+func setupRootless(cmd *cobra.Command, args []string) error {
+	if os.Geteuid() != 0 && cmd != _searchCommand && cmd != _versionCommand && !strings.HasPrefix(cmd.Use, "help") {
```

Comment:
since we are refactoring, could we invert the logic here and do an early return?
```
if os.Geteuid() == 0 || cmd == _searchCommand || cmd == _versionCommand || strings.HasPrefix(cmd.Use, "help")  {
   return nil
}
... the rest of the code with one level less of indentation
```